sdks/python/apache_beam/options/pipeline_options.py

```diff
@@ -362,6 +362,9 @@ def get_all_options(
             'used for internal purposes.' % (','.join(unknown_args)))
       i = 0
       while i < len(unknown_args):
+        # End of argument parsing.
+        if unknown_args[i] == '--':
```

Comment:
Is this related to original intent of this PR or  a new behavior to enable passing args that shouldn't be parsed? 

Can this cause breaking changes?

Reply:
Argparse (as do most other argument parsers) stop parsing anything after a bare `--` as flag values. I added this because I saw a case internally that passed some command line arguments, including a bare `--`, to the options that ended up getting parsed even though they weren't before. 

This shouldn't be a breaking change, if anything it'll make things more consistent.